/*
 * This file was automatically generated by EvoSuite
 * Mon Jan 11 05:08:11 GMT 2021
 */

package com.iluwatar.event.sourcing.event;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import com.iluwatar.event.sourcing.event.AccountCreateEvent;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class AccountCreateEvent_ESTest extends AccountCreateEvent_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      AccountCreateEvent accountCreateEvent0 = new AccountCreateEvent(117L, 1L, 0, (String) null);
      accountCreateEvent0.getOwner();
      assertEquals(0, accountCreateEvent0.getAccountNo());
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      AccountCreateEvent accountCreateEvent0 = new AccountCreateEvent((-1L), 3346L, (-2444), "AccountCreateEvent");
      accountCreateEvent0.getOwner();
      assertEquals((-2444), accountCreateEvent0.getAccountNo());
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      AccountCreateEvent accountCreateEvent0 = new AccountCreateEvent(541L, 541L, 2369, "");
      int int0 = accountCreateEvent0.getAccountNo();
      assertEquals(2369, int0);
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      AccountCreateEvent accountCreateEvent0 = new AccountCreateEvent((-1L), 3346L, (-2444), "AccountCreateEvent");
      int int0 = accountCreateEvent0.getAccountNo();
      assertEquals((-2444), int0);
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      AccountCreateEvent accountCreateEvent0 = new AccountCreateEvent((-1L), 3346L, (-2444), "AccountCreateEvent");
      // Undeclared exception!
      try { 
        accountCreateEvent0.process();
       //  fail("Expecting exception: RuntimeException");
       // Unstable assertion
      } catch(RuntimeException e) {
         //
         // Account already exists
         //
         verifyException("com.iluwatar.event.sourcing.event.AccountCreateEvent", e);
      }
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      AccountCreateEvent accountCreateEvent0 = new AccountCreateEvent(117L, 1L, 0, (String) null);
      int int0 = accountCreateEvent0.getAccountNo();
      assertEquals(0, int0);
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      AccountCreateEvent accountCreateEvent0 = new AccountCreateEvent(704L, 704L, 0, "");
      accountCreateEvent0.getOwner();
      assertEquals(0, accountCreateEvent0.getAccountNo());
  }
}
